src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -28,39 +27,51 @@
           "crumley@google.com",
           "lgmt@google.com");
 
-  public LoginObject verifyToken(HttpServletRequest request) throws IOException {
+  public JSONObject verifyToken(HttpServletRequest request) throws IOException {
     String idTokenString = request.getParameter("token");
-    Boolean checkWhitelist = Boolean.parseBoolean(request.getParameter("private"));
     UrlFetchTransport transport = UrlFetchTransport.getDefaultInstance();
     GsonFactory json = GsonFactory.getDefaultInstance();
     GoogleIdTokenVerifier verifier =
         new GoogleIdTokenVerifier.Builder(transport, json)
             .setAudience(Collections.singletonList(clientId))
             .build();
+
+    JSONObject response = new JSONObject();
+
     try {
       GoogleIdToken idToken = verifier.verify(idTokenString);
       if (idToken != null) {
         Payload payload = idToken.getPayload();
         String userId = payload.getSubject();
-        if (checkWhitelist && !(whitelist.contains(payload.getEmail()))) {
-          return new LoginObject("", false, "NOT AUTHORIZED");
+        response.put("userId", userId);
+        response.put("verified", "true");
+        if (whitelist.contains(payload.getEmail())) {
+          response.put("whitelist", true);
```

Comment:
FYI as a community we're trying to remove references to white/blacklist and instead use allowlist and blocklist. 
Might be good to add at least a TODO to that note. Ideally something to refactor before we send out code (it's a Google mandate now I think that all new code should be properly renamed)

Reply:
Oh thanks for pointing that out! Will do right now